### PR TITLE
Review Page boilerplate

### PR DESCRIPTION
**Pull request by michaelbayday**
Review Page boilerplate

Issue: gitdiffparser is not liking some of our diff data

image

### CODE REVIEW

File: components/ChallengeMaterial.tsx [hunk 1]
```diff
@@ -139,7 +139,11 @@ export const ChallengeTitleCard: React.FC<ChallengeTitleCardProps> = ({
 export const ChallengeQuestionCard: React.FC<ChallengeQuestionCardProps> = ({
   currentChallenge
 }) => {
-  const diff = _.get(currentChallenge, 'submission.diff', false)
+  //temp fix while CLI tool is getting fixed
```
Comment (songz) [inline components/ChallengeMaterial.tsx:142]: could use a comment on what the regex is filtering out for. In the industry, regexes should always be accompanied by a comment.

File: graphql/queries.ts [hunk 2]
```diff
@@ -99,10 +110,7 @@ export const GET_APP = gql`
         order
         challengeId
         lessonId
-        reviewer {
-          id
-          username
```
Comment (songz) [inline graphql/queries.ts:104]: don't you want to view who reviewed your submission? I think reviewer id and username is important.

File: graphql/queries.ts [hunk 3]
```diff
@@ -61,7 +61,18 @@ export const SIGNUP_USER = gql`
...
+      id
       status
+      diff
+      comment
+      challengeId
+      user {
+        id
+        username
+      }
+      reviewerId
```
Comment (songz) [inline graphql/queries.ts:73]: same should this be reviewer instead?

File: helpers/controllers/submissionController.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ type ArgsCreateSubmission = {
   challengeId: string
 }
 
-type Submission = {
+type ArgsGetSubmissions = {
```
Comment (songz) [inline helpers/controllers/submissionController.ts:15]: I think a better name: `GetSubmissionsWhere`

File: helpers/controllers/submissionController.ts [hunk 5]
```diff
@@ -23,12 +23,14 @@ export const createSubmission = async (
   try {
     if (!args) throw new Error('Invalid args')
     const { challengeId, cliToken, diff, lessonId } = args
+    //to revert once CLI is fixed
+    const fixedDiff = diff.replace(/(.?\[\d*m)/g, '')
```
Comment (songz) [inline helpers/controllers/submissionController.ts:27]: if you do this here, why do it on the front end?

File: helpers/controllers/submissionController.ts [hunk 6]
```diff
@@ -43,12 +45,53 @@ export const createSubmission = async (
...
+  const subs = await Submission.findAll({
     where: {
       status: 'open',
       lessonId
     }
   })
+  if (subs.length === 0) {
+    return subs
+  }
+  const userIds = subs.reduce((acc: Set<number>, sub: any) => {
```
Comment (songz) [inline helpers/controllers/submissionController.ts:63]: none of these should be necessary, you just use an `include` field in your submission query.

File: helpers/controllers/submissionController.ts (hunk 5 above)
Comment (michaelbayday) [reply helpers/controllers/submissionController.ts:27]: This is for handling new submissions for the interim until CLi is fixed

File: graphql/queries.ts (hunk 2 above)
Comment (michaelbayday) [reply graphql/queries.ts:104]: good point haha

**Comment by songz** [comment]
HOT MERGE

**Pull request merged**